rfcs/75-fragmented-levels.md

```diff
@@ -0,0 +1,141 @@
+fn load_region(
+    events: EventReader<Region>,
+    level_manager: ResMut<LevelAPI>,
+) {
+    for region in events.iter() {
+        match region {
+            Region::EnterCave => level_manager.load_scene("mysterious cave"),
+            Region::LeaveCave => level_manager.unload_scene("mysterious cave"),
+            Region::EnterMansion => level_manager.unload_all_scenes().load_scene("mansion interior"),
+            Region::LeaveMansion => level_manager.unload_all_scenes().load_scene("large forest north"),
+        }
+    }
+}
+
+// Example of changing zones/levels in an MMO like Albion Online or Lost Ark.
+fn change_zone(
+    events: EventReader<Zone>,
+    level_manager: ResMut<LevelAPI>,
+) {
+    for zone in events.iter() {
+        match zone {
+            // .change_level would unload all the scenes belonging to the current level,
+            // then load in all or only specificed scenes from the new level.
+
+            // Loads specific scenes from level
+            Zone::LazzyForest => level_manager.change_level("levels/lazzy_forest.level",
+                "north-outer-road",
+                "north-outer-town",
+                "north-outer-forest"
+            ),
+            // Loads all scenes from level
+            Zone::LakeCastleMount => level_manager.change_level("levels/lake_castle_mount.level");
+        }
+    }
+}
+```
+
+### Does This Make Bevy Scene's Complicated?
+No, scenes are separate to levels and don't know what levels are. Levels are an opt in feature of Bevy. If you are making an application that doesn't need levels, don't use them. If you want levels, but also want to load some scenes manually without using levels that's also fine.
+
+If a scene is loaded through the level bevy resource, then it's managed by the level resource. If you load a scene without using the level resource, then it's external and the levels API won't control it.
+
+
+
+## Implementation strategy
+
+`fragmented-levels` would be implemented in a crate called `bevy_level`. (I reserved the name for this encase the RFC is accepted and it's needed [https://crates.io/crates/bevy_level](https://crates.io/crates/bevy_level), if cart/bevy wants it let me know and I can transfer the name.)
+
+The level API would be interacted with via a bevy resource, and only 1 level can be loaded at any given time. API calls on the resource would go to a list, and then get processed by systems in the `bevy_level` crate.
+
+The level format and API can be implemented independantly of scenes and ECS world. ECS world and scenes don't understand or have any dependancies on level. Because of this levels are an opt in feature that are not be part of the default plugin.
+
+Because of this ECS world and scenes can be worked on indipendantly from levels without breaking anything.
+
+### Level File Format
+todo
+
+
+
+## Drawbacks
+
+- It's an extra crate to maintain.
+
+
+## Rationale and alternatives
```

Comment:
Could add to this section what differentiates the strategy proposed in this RFC from something like nested scenes (which, to my knowledge, is a [planned feature](https://github.com/bevyengine/bevy/discussions/9538))? It feels like a level could also be created as a scene of scenes.

Reply:
I read about that a week or two ago and completely forgot where I had heard about scene nesting. Then for some dumb reason assumed it was a UI only thing.

I'll amend the RFC at some point to use nested scenes to replace level files. Using nested scenes is also an improvement since you can have multiple nested scenes with layers without making the API more complicated because multiple nested scenes are planned on being supported already.

Reply:
Just realized one side affect of used the new bevy scenes file instead of the level file, it will make the new scene API and file format more complicated.

When you load a level, you don't want to always load all the attached scenes. If I'm not mistaken the current plan is for a scene when opened is to load itself and all nested scenes recursively. The level API doesn't do this. Instead it gives a handle, and you specify which scenes you want loaded along with their visibility and/or transparency.

If this uses the new bevy scene, then changes may need to be made to how bevy scene is intended to work. Which would make this an extension RFC for bevy scenes.

Personally I think keeping `Fragmented-Level` and bevy scenes disconnect is better than making it a feature of bevy scenes. But not using scenes since they support nesting seems like a shame, so I'm not sure which is better. Want some more opinions on this.